02.calendar/cal.rb

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require 'date'
+
+def generate_calender
+  params = ARGV.getopts('m:', 'y:')
+
+  month = params['m'].to_i.between?(1, 12) ? params['m'].to_i : Date.today.month
+  year = params['y'].to_i.between?(1970, 2100) ? params['y'].to_i : Date.today.year
+
+  first_date = Date.new(year, month, 1)
+  last_date = Date.new(year, month, -1)
+
+  calender_dates = (first_date..last_date).map { |date| date == Date.today ? "\e[30m\e[47m#{date.day.to_s.rjust(2)}\e[0m" : date.day.to_s.rjust(2) }
```

Comment:
横に長すぎるので改行を入れましょう

```suggestion
  calender_dates = (first_date..last_date).map do |date|
    date == Date.today ? "\e[30m\e[47m#{date.day.to_s.rjust(2)}\e[0m" : date.day.to_s.rjust(2)
  end
```

https://twitter.com/jnchito/status/1326468865103073280

Reply:
> 「書いたコードが動いた＝文法的に問題がないからOK」じゃなくて、「自分以外の誰かが見てもぱっと理解できるか？」も常に自問しましょう。

共有ありがとうございます！「自分以外の誰かが見てもぱっと理解できるか？」まで考えて実装できていなかったので、次プラクティスからはしっかり考えて実装していきます！